src/ray/common/ray_config_def.h

```diff
@@ -768,19 +768,31 @@ RAY_CONFIG(int64_t, grpc_keepalive_time_ms, 10000)
 /// grpc keepalive timeout.
 RAY_CONFIG(int64_t, grpc_keepalive_timeout_ms, 20000)
 
-/// NOTE: we set a loose client keep alive because
+/// NOTE: We set a loose client keep alive because
 /// they have a failure model that considers network failures as component failures
 /// and this configuration break that assumption. We should apply to every other component
 /// after we change this failure assumption from code.
 /// grpc keepalive timeout for client.
+/// TODO this should be lowered to 30s after
+/// https://github.com/ray-project/ray/issues/29656
 RAY_CONFIG(int64_t, grpc_client_keepalive_time_ms, 300000)
-
-/// grpc keepalive timeout for client.
+/// gRPC keep-alive timeout for client.
+/// TODO this should be lowered to 15s after
```

Comment:
why is that?

Reply:
Because 5m period with 2m timeout doesn't make sense. 

We'd test more frequently and timeout of more than 15s for a single PING frame doesn't make sense.

Reply:
How do you come up with the new 30s and 15s. Why these two specific values?

Also I looked at the history of why we picked this value initially: https://github.com/ray-project/ray/pull/31640/files#r1068600835. If we change `grpc_client_keepalive_time_ms` we need to change the server config `GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS` as well.

Let's remove these comments for now so this PR is purely about adding keep-alive configs. We can debate on the other configs in a separate PR.

Reply:
> How do you come up with the new 30s and 15s. Why these two specific values?

We need to balance speed of detection with the overhead of sending PING frames over the transport. 0.5Hz is low enough frequency to not overburden the transport but low enough to be able to detect disconnects w/in 30s. 

15s timeout should be sufficient for any co-located (w/in reasonable enough proximity of the physical nodes) deployed cluster. For those clusters deployed x-region/x-cloud (which there are likely not as many of) folks could tune this up to their own optimal settings.

> If we change grpc_client_keepalive_time_ms we need to change the server config GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS as well.

I don't think we need to change that value (went t/h that thread)

Reply:
I'm ok removing the values, but i'd prefer to keep the TODO to make sure it's not slipping once we resolve ref'd issue